Refactor

jankeromnes opened the pull request:
Hello! I refactored a few things because copy/pasta made me sad. Also suggested some (in my opinion) less confusing names.

@espadrine let me know what you think!

jankeromnes commented on lib/camp.js:
```diff
@@ -112,7 +112,7 @@ function jsonFromQuery(query, obj) {
 // - application/x-www-form-urlencoded
 // - multipart/form-data
-function getQueries(ask, end) {
+function parseQuery(ask, end) {
```
In the comment, you say "parse the query", and there seems to ever only be one "query".

espadrine replied:
There's the x-www-form-urlencoded query and the multipart query.

jankeromnes replied:
Isn't it either x-www-form-urlencoded data or multipart data?

espadrine replied:
Nah, they override one another.

jankeromnes commented:
What I did:

- Created "generic routing unit" so that both `routeUnit` and `notfoundUnit` could use it.
- Folded `catchpath` into it, because it was the only remaining callsite, and the name was confusing (I thought it was another unit but with different params).
- Extracted the deflate/gzip code into a `compressed(ask)` function which returns a stream.

jankeromnes commented on lib/camp.js:
```diff
@@ -492,149 +506,103 @@ function staticUnit (server) {
+      filteredRoutes.forEach (function (path) { warnMsg += '- ' + path[0]; });
+      log(warnMsg, 'warn');
+    }
+    var pathmatch = ask.path.match (RegExp (filteredRoutes[0][0]));
```
Whoops (I changed the route structure to `{pattern, handler}` object).

jankeromnes commented on lib/camp.js:
```diff
@@ -492,149 +506,103 @@ function staticUnit (server) {
+          , 'error');
+      } else {
+        // params: template parameters (JSON-serializable).
+        filteredRoutes[0][1](ask.query, pathmatch, function end (params, options) {
```
Whoops.

jankeromnes commented on lib/camp.js:
```diff
@@ -492,149 +506,103 @@ function staticUnit (server) {
-        }
+    if (filteredRoutes.length > 1) {
+      var warnMsg = 'More than one template path match' + ask.path + ':\n';
+      filteredRoutes.forEach (function (path) { warnMsg += '- ' + path[0]; });
```
Whoops.

espadrine commented on lib/camp.js:
```diff
@@ -484,157 +500,110 @@ function staticUnit (server) {
-
-  server.route = route;
+// Generic routing unit.
+function routingUnit (server, name) {
```
`routingEngine`

jankeromnes commented:
Addressed some nits, but let's take smaller steps instead of compulsively refactoring everything. I'll open smaller separate pull requests which will be safer to review and accept.
